runtime/src/accounts_db.rs

```diff
@@ -3014,15 +3018,18 @@ impl AccountsDb {
 
         let mut alive = 0;
         let mut dead = 0;
-
-        accounts[..std::cmp::min(accounts.len(), count)]
-            .iter()
-            .for_each(|pair| {
-                let pubkey = &pair.0;
-                let stored_account = &pair.1;
-                let lookup = self.accounts_index.get_account_read_entry(pubkey);
-                if let Some(locked_entry) = lookup {
-                    let is_alive = locked_entry.slot_list().iter().any(|(_slot, acct_info)| {
+        let mut index = 0;
+        self.accounts_index.scan(
+            accounts[..std::cmp::min(accounts.len(), count)]
+                .iter()
+                .map(|(key, _)| key),
+            // return true if we want this item to remain in the cache
```

Comment:
nit: Since `result` is of type `AccountsIndexScanResult`, return `KeepInMemory` (or `Unref`??) to remain in the cache.

This can be handled in a subsequent PR.